Automattic/wp-calypso
# A4A > Migrations: Initial setup of tracking site migrations

Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/1434
Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/1435
Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/1440

## Proposed Changes

This PR:

- Adds the required sections, feature flags, pages, controller, etc
- Updates the permission to allow migration sub-routes.
- Adds a nested side menu for Migrations.
- Renders the existing Payment Settings page from Referrals in Migrations > Payment Settings.

## Why are these changes being made?

* To do the initial setup for tracking site migrations.

## Testing Instructions

* Open the A4A live link > Append the URL with `?flags=-a4a-tracking-site-migrations` and verify that the Migrations screen matches with the production version.
* Remove the feature flag > Refresh the page > Verify the sidebar menu for Migrations is now nested as shown below:

<img width="266" alt="Screenshot 2024-11-08 at 4 29 02 PM" src="https://github.com/user-attachments/assets/a681d21e-3486-4bad-bed0-ca6d067c86e4">

Click the Migrations screen and verify that the screen looks as shown below. Also, verify that clicking on sub-menu items changes the route, and the `Payment Settings` option renders the Referrals' payment settings page.

<img width="996" alt="Screenshot 2024-11-08 at 4 29 07 PM" src="https://github.com/user-attachments/assets/abf4ea4e-4d5b-4d70-a178-26d957aad757">

## Pre-merge Checklist
